sql/catalog/lease: TestLeaseManagerSessionIDChanges failed

sql/catalog/lease.TestLeaseManagerSessionIDChanges [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19631527?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19631527?buildTab=artifacts#/) on master @ [cab3b7f31e202a4e6e94d9ba901cccbdc47b25e3](https://github.com/cockroachdb/cockroach/commits/cab3b7f31e202a4e6e94d9ba901cccbdc47b25e3):


```
=== RUN   TestLeaseManagerSessionIDChanges
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/be0d053872b35521d816dbd3d2572df0/logTestLeaseManagerSessionIDChanges746944333
    test_log_scope.go:76: use -show-logs to present logs inline
    lease_internal_test.go:1722: query 'SELECT session_id FROM system.lease WHERE desc_id=106': expected:
        session-1
        
        got:
        ����"bζDK}����=��f
        session-1
        
    panic.go:635: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/be0d053872b35521d816dbd3d2572df0/logTestLeaseManagerSessionIDChanges746944333
--- FAIL: TestLeaseManagerSessionIDChanges (0.52s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseManagerSessionIDChanges.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-50436

--- a/pkg/sql/catalog/lease/lease_internal_test.go
+++ b/pkg/sql/catalog/lease/lease_internal_test.go
@@ -1653,7 +1653,11 @@ func TestLeaseManagerSessionIDChanges(t *testing.T) {
 	runner := sqlutils.MakeSQLRunner(sqlDB)
 	runner.Exec(t, `SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false`)
 	runner.Exec(t, `SET CLUSTER SETTING sql.catalog.descriptor_wait_for_initial_version.enabled = false`)
-
+	// Disable the range feed for this test. This prevents a race condition
+	// where the range feed could insert a lease with the real session ID
+	// (if the session ID hook was disabled) before the initial descriptor
+	// acquisition completes.
+	leaseManager.TestingSetDisableRangeFeedCheckpointFn(true)
 	runner.Exec(t, `
 CREATE DATABASE t;
 CREATE TABLE t.test (k CHAR PRIMARY KEY, v CHAR);
@@ -1695,6 +1699,7 @@ CREATE TABLE t.test (k CHAR PRIMARY KEY, v CHAR);
 		// Restore the original provider so valid session IDs
 		// are assigned.
 		leaseManager.storage.livenessProvider = fs.Provider
+		leaseManager.TestingSetDisableRangeFeedCheckpointFn(false)
 	}()
 
 	// Repeatedly lease the same descriptor, with the session ID continuously changing
@@ -1713,7 +1718,7 @@ CREATE TABLE t.test (k CHAR PRIMARY KEY, v CHAR);
 		// to be expired.
 		newSessionID := getLatestLeasedDesc().getSessionID()
 		newExpiry := getLatestLeasedDesc().getExpiration(ctx)
-		require.NotEqualf(t, previousSessionID, newSessionID, "session ID should not match")
+		require.NotEqualf(t, previousSessionID, newSessionID, "session ID should not match %v %v", previousSessionID, newSessionID)
 		require.Truef(t, previousExpiry.Less(newExpiry), "session expiry should be later.")
 		// Disable the hook before the lease query.
 		enableHook.Swap(false)

